proto/ibc/lightclients/wasm/v1/tx.proto

```diff
@@ -27,3 +30,19 @@ message MsgStoreCodeResponse {
   // the sha256 hash of the stored code
   bytes checksum = 1;
 }
+
+// MsgMigrateContract defines the request type for the MigrateContract rpc.
+message MsgMigrateContract {
+  option (cosmos.msg.v1.signer) = "signer";
+
+  string signer = 1;
+  // the client id of the contract
+  string client_id = 2;
+  // the code hash of the new wasm byte code for the contract
+  bytes new_code_hash = 3;
```

Comment:
I would just call this `code_hash` and skip `new`, just like [they do in `x/wasm`](https://github.com/CosmWasm/wasmd/blob/main/proto/cosmwasm/wasm/v1/tx.proto#L215).

Reply:
I initially called it that but changed based on @colin-axner 's [suggestion here](https://github.com/cosmos/ibc-go/pull/5000#discussion_r1378669084).

Reply:
I also prefer `code_hash`, but I am ok to keep `new_code_hash` if Colin feels strongly about it. Fine merging with either and changing it in a follow-up PR if needed.

Reply:
reverted back to `code_hash`. Consensus by @srdtrk, @crodriguezvega, @charleenfei